### PR TITLE
bcc: 0.28.0 -> 0.29.1

### PR DESCRIPTION
**Pull request by mfrw**
Signed-off-by: Muhammad Falak R Wani <falakreyaz@gmail.com>

## Description of changes
Diff: https://github.com/iovisor/bcc/compare/v0.28.0...v0.29.1

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/os-specific/linux/bcc/default.nix [hunk 1]
```diff
@@ -20,15 +20,15 @@
...
+  version = "0.29.1";
 
   disabled = !stdenv.isLinux;
 
   src = fetchFromGitHub {
     owner = "iovisor";
     repo = "bcc";
     rev = "v${version}";
-    sha256 = "sha256-+ecSaVroDC2bWbio4JsuwEvHQdCMpxLt7hIkeREMJs8=";
+    sha256 = "sha256-+HYCweAI5axx0ZNFd/jLRXkUinRLDmKWMpLTk7FrEe0=";
```
Comment (gepbird) [inline pkgs/os-specific/linux/bcc/default.nix:31]:
```suggestion
    hash = "sha256-+HYCweAI5axx0ZNFd/jLRXkUinRLDmKWMpLTk7FrEe0=";
```

**Review by gepbird — CHANGES_REQUESTED** [review]

**Comment by gepbird** [comment]
Result of `nixpkgs-review pr 272951` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>picosnitch</li>
    <li>picosnitch.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>bcc (linuxKernel.packages.linux_4_19.bcc ,linuxKernel.packages.linux_4_19_hardened.bcc ,linuxKernel.packages.linux_5_10.bcc ,linuxKernel.packages.linux_5_10_hardened.bcc ,linuxKernel.packages.linux_5_15.bcc ,linuxKernel.packages.linux_5_15_hardened.bcc ,linuxKernel.packages.linux_5_4.bcc ,linuxKernel.packages.linux_5_4_hardened.bcc ,linuxKernel.packages.linux_6_1.bcc ,linuxKernel.packages.linux_6_1_hardened.bcc ,linuxKernel.packages.linux_6_5.bcc ,linuxKernel.packages.linux_6_5_hardened.bcc ,linuxKernel.packages.linux_6_6.bcc ,linuxKernel.packages.linux_hardened.bcc ,linuxKernel.packages.linux_latest_libre.bcc ,linuxKernel.packages.linux_libre.bcc ,linuxKernel.packages.linux_lqx.bcc ,linuxKernel.packages.linux_testing_bcachefs.bcc ,linuxKernel.packages.linux_xanmod.bcc ,linuxKernel.packages.linux_xanmod_latest.bcc ,linuxKernel.packages.linux_xanmod_stable.bcc ,linuxKernel.packages.linux_zen.bcc ,linuxPackages-libre.bcc ,linuxPackages.bcc ,linuxPackages_4_19_hardened.bcc ,linuxPackages_5_10_hardened.bcc ,linuxPackages_5_15_hardened.bcc ,linuxPackages_5_4_hardened.bcc ,linuxPackages_6_1_hardened.bcc ,linuxPackages_6_5_hardened.bcc ,linuxPackages_hardened.bcc ,linuxPackages_latest-libre.bcc ,linuxPackages_latest.bcc ,linuxPackages_lqx.bcc ,linuxPackages_testing_bcachefs.bcc ,linuxPackages_xanmod.bcc ,linuxPackages_xanmod_latest.bcc ,linuxPackages_xanmod_stable.bcc ,linuxPackages_zen.bcc)</li>
    <li>bcc.man (linuxKernel.packages.linux_4_19.bcc.man ,linuxKernel.packages.linux_4_19_hardened.bcc.man ,linuxKernel.packages.linux_5_10.bcc.man ,linuxKernel.packages.linux_5_10_hardened.bcc.man ,linuxKernel.packages.linux_5_15.bcc.man ,linuxKernel.packages.linux_5_15_hardened.bcc.man ,linuxKernel.packages.linux_5_4.bcc.man ,linuxKernel.packages.linux_5_4_hardened.bcc.man ,linuxKernel.packages.linux_6_1.bcc.man ,linuxKernel.packages.linux_6_1_hardened.bcc.man ,linuxKernel.packages.linux_6_5.bcc.man ,linuxKernel.packages.linux_6_5_hardened.bcc.man ,linuxKernel.packages.linux_6_6.bcc.man ,linuxKernel.packages.linux_hardened.bcc.man ,linuxKernel.packages.linux_latest_libre.bcc.man ,linuxKernel.packages.linux_libre.bcc.man ,linuxKernel.packages.linux_lqx.bcc.man ,linuxKernel.packages.linux_testing_bcachefs.bcc.man ,linuxKernel.packages.linux_xanmod.bcc.man ,linuxKernel.packages.linux_xanmod_latest.bcc.man ,linuxKernel.packages.linux_xanmod_stable.bcc.man ,linuxKernel.packages.linux_zen.bcc.man ,linuxPackages-libre.bcc.man ,linuxPackages.bcc.man ,linuxPackages_4_19_hardened.bcc.man ,linuxPackages_5_10_hardened.bcc.man ,linuxPackages_5_15_hardened.bcc.man ,linuxPackages_5_4_hardened.bcc.man ,linuxPackages_6_1_hardened.bcc.man ,linuxPackages_6_5_hardened.bcc.man ,linuxPackages_hardened.bcc.man ,linuxPackages_latest-libre.bcc.man ,linuxPackages_latest.bcc.man ,linuxPackages_lqx.bcc.man ,linuxPackages_testing_bcachefs.bcc.man ,linuxPackages_xanmod.bcc.man ,linuxPackages_xanmod_latest.bcc.man ,linuxPackages_xanmod_stable.bcc.man ,linuxPackages_zen.bcc.man)</li>
    <li>bpftrace (linuxKernel.packages.linux_4_19.bpftrace ,linuxKernel.packages.linux_4_19_hardened.bpftrace ,linuxKernel.packages.linux_5_10.bpftrace ,linuxKernel.packages.linux_5_10_hardened.bpftrace ,linuxKernel.packages.linux_5_15.bpftrace ,linuxKernel.packages.linux_5_15_hardened.bpftrace ,linuxKernel.packages.linux_5_4.bpftrace ,linuxKernel.packages.linux_5_4_hardened.bpftrace ,linuxKernel.packages.linux_6_1.bpftrace ,linuxKernel.packages.linux_6_1_hardened.bpftrace ,linuxKernel.packages.linux_6_5.bpftrace ,linuxKernel.packages.linux_6_5_hardened.bpftrace ,linuxKernel.packages.linux_6_6.bpftrace ,linuxKernel.packages.linux_hardened.bpftrace ,linuxKernel.packages.linux_latest_libre.bpftrace ,linuxKernel.packages.linux_libre.bpftrace ,linuxKernel.packages.linux_lqx.bpftrace ,linuxKernel.packages.linux_testing_bcachefs.bpftrace ,linuxKernel.packages.linux_xanmod.bpftrace ,linuxKernel.packages.linux_xanmod_latest.bpftrace ,linuxKernel.packages.linux_xanmod_stable.bpftrace ,linuxKernel.packages.linux_zen.bpftrace ,linuxPackages-libre.bpftrace ,linuxPackages.bpftrace ,linuxPackages_4_19_hardened.bpftrace ,linuxPackages_5_10_hardened.bpftrace ,linuxPackages_5_15_hardened.bpftrace ,linuxPackages_5_4_hardened.bpftrace ,linuxPackages_6_1_hardened.bpftrace ,linuxPackages_6_5_hardened.bpftrace ,linuxPackages_hardened.bpftrace ,linuxPackages_latest-libre.bpftrace ,linuxPackages_latest.bpftrace ,linuxPackages_lqx.bpftrace ,linuxPackages_testing_bcachefs.bpftrace ,linuxPackages_xanmod.bpftrace ,linuxPackages_xanmod_latest.bpftrace ,linuxPackages_xanmod_stable.bpftrace ,linuxPackages_zen.bpftrace)</li>
    <li>bpftrace.man (linuxKernel.packages.linux_4_19.bpftrace.man ,linuxKernel.packages.linux_4_19_hardened.bpftrace.man ,linuxKernel.packages.linux_5_10.bpftrace.man ,linuxKernel.packages.linux_5_10_hardened.bpftrace.man ,linuxKernel.packages.linux_5_15.bpftrace.man ,linuxKernel.packages.linux_5_15_hardened.bpftrace.man ,linuxKernel.packages.linux_5_4.bpftrace.man ,linuxKernel.packages.linux_5_4_hardened.bpftrace.man ,linuxKernel.packages.linux_6_1.bpftrace.man ,linuxKernel.packages.linux_6_1_hardened.bpftrace.man ,linuxKernel.packages.linux_6_5.bpftrace.man ,linuxKernel.packages.linux_6_5_hardened.bpftrace.man ,linuxKernel.packages.linux_6_6.bpftrace.man ,linuxKernel.packages.linux_hardened.bpftrace.man ,linuxKernel.packages.linux_latest_libre.bpftrace.man ,linuxKernel.packages.linux_libre.bpftrace.man ,linuxKernel.packages.linux_lqx.bpftrace.man ,linuxKernel.packages.linux_testing_bcachefs.bpftrace.man ,linuxKernel.packages.linux_xanmod.bpftrace.man ,linuxKernel.packages.linux_xanmod_latest.bpftrace.man ,linuxKernel.packages.linux_xanmod_stable.bpftrace.man ,linuxKernel.packages.linux_zen.bpftrace.man ,linuxPackages-libre.bpftrace.man ,linuxPackages.bpftrace.man ,linuxPackages_4_19_hardened.bpftrace.man ,linuxPackages_5_10_hardened.bpftrace.man ,linuxPackages_5_15_hardened.bpftrace.man ,linuxPackages_5_4_hardened.bpftrace.man ,linuxPackages_6_1_hardened.bpftrace.man ,linuxPackages_6_5_hardened.bpftrace.man ,linuxPackages_hardened.bpftrace.man ,linuxPackages_latest-libre.bpftrace.man ,linuxPackages_latest.bpftrace.man ,linuxPackages_lqx.bpftrace.man ,linuxPackages_testing_bcachefs.bpftrace.man ,linuxPackages_xanmod.bpftrace.man ,linuxPackages_xanmod_latest.bpftrace.man ,linuxPackages_xanmod_stable.bpftrace.man ,linuxPackages_zen.bpftrace.man)</li>
    <li>oci-seccomp-bpf-hook (linuxPackages.oci-seccomp-bpf-hook)</li>
    <li>oci-seccomp-bpf-hook.man (linuxPackages.oci-seccomp-bpf-hook.man)</li>
    <li>linuxPackages.system76-scheduler (linuxPackages-libre.system76-scheduler ,linuxPackages_4_19_hardened.system76-scheduler ,linuxPackages_5_10_hardened.system76-scheduler ,linuxPackages_5_15_hardened.system76-scheduler ,linuxPackages_5_4_hardened.system76-scheduler ,linuxPackages_6_1_hardened.system76-scheduler ,linuxPackages_6_5_hardened.system76-scheduler ,linuxPackages_hardened.system76-scheduler ,linuxPackages_latest-libre.system76-scheduler ,linuxPackages_latest.system76-scheduler ,linuxPackages_lqx.system76-scheduler ,linuxPackages_testing_bcachefs.system76-scheduler ,linuxPackages_xanmod.system76-scheduler ,linuxPackages_xanmod_latest.system76-scheduler ,linuxPackages_xanmod_stable.system76-scheduler ,linuxPackages_zen.system76-scheduler)</li>
    <li>sockdump</li>
  </ul>
</details>

**Review by martinetd — APPROVED** [review]
(sorry for the delay, I was waiting for ofborg tests the other day and forgot about it, I wonder if there's a way to subscribe to some specific check results?... given they're added dynamically when it's over I doubt it :/)

Anyway:
 - picosnitch failure seems unrelated at glance,  does it build before this PR?
 - the ofborg aarch64 test fails because of a failing python3.11-sphinx test which makes the derivation fail and it's a dependency of half the world; so we still have no idea if aarch64 works :shrug: (saying this because we gave up on aarch64 build timing out an libbpf upgrade a couple of weeks ago)

Anyway, change looks good and x86_64 test is happy, so probably good enough.

**Comment by gepbird** [comment]
picosnitch fails to build before this PR too so yes, it's unrelated.

**Review by gepbird — APPROVED** [review]

**Pull request merged**